api/src/tests/resource_groups.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use super::new_test_context;
+use aptos_api_test_context::current_function_name;
+use aptos_cached_packages::aptos_stdlib;
+use aptos_framework::BuiltPackage;
+use std::path::PathBuf;
+
+// This test verifies that both READ APIs can seamlessly translate from resource group to resource
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+async fn test_read_resoure_group() {
+    let mut context = new_test_context(current_function_name!());
+
+    let mut account = context.gen_account();
+    let txn = context.create_user_account(&account);
+    let bcs_txn = bcs::to_bytes(&txn).unwrap();
+    context
+        .expect_status_code(202)
+        .post_bcs_txn("/transactions", bcs_txn)
+        .await;
+    context.commit_mempool_txns(1).await;
+
+    let request = format!("/accounts/{}/resources", account.address());
+    let resource_count = context.get(&request).await.as_array().unwrap().len();
+
+    let path = PathBuf::from(std::env!("CARGO_MANIFEST_DIR"))
+        .join("../aptos-move/move-examples/resource_groups/primary");
+    let mut build_options = aptos_framework::BuildOptions::default();
+    build_options
+        .named_addresses
+        .insert("resource_groups_primary".to_string(), account.address());
+
+    let package = BuiltPackage::build(path.to_owned(), build_options).unwrap();
+    let code = package.extract_code();
+    let metadata = package.extract_metadata().unwrap();
+
+    let payload = aptos_stdlib::code_publish_package_txn(bcs::to_bytes(&metadata).unwrap(), code);
```

Comment:
Can you add some comments to explain what the test is testing, how things are setup, and why the expected result makes sense? This would make it easier for unfamiliar eyes to debug failures in the future

Reply:
Sure!

The gist is we publish a resource group package with an init module and then read the resource group data in the two different ways that the API allows for reading resources.